### PR TITLE
HT-2524 cloud storage audit

### PR DESCRIPTION
**Pull request by moseshll**
Notes:
1) Minio does not fully support `restore-object`/Glacier so there are some hacks in place in the testsuite.
2) As with the Data Den audit code, there is an "evil hack" note regarding storage config.
3) Added DB timestamp column is called `feed_backups.restore_request`

### CODE REVIEW

**Comment by aelkiss** [comment]
Sorry I haven't been able to get to this this sprint -- hope to get back to it soon.

**Comment by moseshll** [comment]
Rebasing to pick up changes from HT-2812 and continuing from there with more commits or a new branch.

File: lib/HTFeed/GlacierZipAudit.pm [hunk 1]
```diff
@@ -32,39 +33,39 @@ sub choose {
...
 
 # Get the objects for which we have issued restore requests
 sub pending_objects {
   my $path_prefix = shift || 's3://ht-deep-archive-backup/';
 
   my $dbh = HTFeed::DBTools->get_dbh();
-  my ($namespace, $objid, $path, $version);
-  my $sql = 'SELECT namespace,id,path,version FROM feed_backups' .
+  my $sql = 'SELECT namespace,id,path,version,storage_name FROM feed_backups' .
             " WHERE path LIKE '$path_prefix%'" .
```
Comment (aelkiss) [inline lib/HTFeed/GlacierZipAudit.pm]: I think this is the main thing we'll want to change to condition on `storage_name` instead of `$path_prefix`

Comment (moseshll) [reply lib/HTFeed/GlacierZipAudit.pm]: `storage_name` set to `s3-deep-archive-mlibrary` by default, or would you prefer any storage name kept out of the codebase entirely and only passed in from `bin/some_script.pl`?

Comment (aelkiss) [reply lib/HTFeed/GlacierZipAudit.pm]: Let's keep it out of the codebase entirely and pass it as an argument to `bin/some_script.pl`

**Review by aelkiss — APPROVED** [review]
Changes to use storage_name looks good. I will go ahead and merge.

**Pull request merged**